src/huggingface_hub/hf_api.py

```diff
@@ -2245,6 +2264,674 @@ def get_full_repo_name(
+
+        Returns:
+            `Pagination[Discussion]`: A [`Pagination`] of [`Discussion`].
+        """
+        if repo_type not in REPO_TYPES:
+            raise ValueError(f"Invalid repo type, must be one of {REPO_TYPES}")
+        if repo_type is None:
+            repo_type = REPO_TYPE_MODEL
+        repo_id = f"{repo_type}s/{repo_id}"
+        if token is None:
+            token = HfFolder.get_token()
+
+        path = f"{self.endpoint}/api/{repo_id}/discussions"
+
+        resp = requests.get(
+            path,
+            headers={"Authorization": f"Bearer {token}"} if token else None,
+        )
+        _raise_for_status(resp)
+
+        paginated_discussions = resp.json()
+        discussions = paginated_discussions["discussions"]
+        total = paginated_discussions["count"]
+        start = paginated_discussions["start"]
+
+        has_next = (start + len(discussions)) < total
+
+        next_page = (
+            partial(
+                self.get_repo_discussions,
+                repo_id=repo_id,
+                repo_type=repo_type,
+                token=token,
+                _page_index=_page_index + 1,
+            )
+            if has_next
+            else None
+        )
+
+        return Pagination(
+            page=[
+                Discussion(
+                    title=discussion["title"],
+                    num=discussion["num"],
+                    author=discussion.get("author", {}).get("name", "deleted"),
+                    created_at=parse_datetime(discussion["createdAt"]),
+                    status=discussion["status"],
+                    repo_id=discussion["repo"]["name"],
+                    repo_type=discussion["repo"]["type"],
+                    is_pull_request=discussion["isPullRequest"],
+                )
+                for discussion in discussions
+            ],
+            page_num=_page_index,
+            total=total,
+            next_page=next_page,
+        )
+
+    def get_discussion_details(
+        self,
+        repo_id: str,
+        discussion_num: int,
+        *,
+        repo_type: Optional[str] = None,
+        token: Optional[str] = None,
+    ) -> DiscussionWithDetails:
+        """Fetches a discussion's / pull request's details from the Hub
+
+        Args:
+            repo_id (`str`):
+                A namespace (user or an organization) and a repo name separated
+                by a `/`.
+            discussion_num (`int`):
+                The number of the discussion or pull request. Must be a strictly positive integer.
+            repo_type (`str`, *optional*):
+                Set to `"dataset"` or `"space"` if uploading to a dataset or
+                space, `None` or `"model"` if uploading to a model. Default is
+                `None`.
+            token (`str`, *optional*):
+                An authentication token (See https://huggingface.co/settings/token)
+
+        Returns: [`DiscussionWithDetails`]
+
+        <Tip>
```

Comment:
There is a dedicated `Raises` section in the standard we use for docs, they don't have to be in a `Tip` tag.

Reply:
I think quite a few of those are recent and added by you maybe? I remember adding quite a few just under `Raises`. The point here would be how easy it is for a future developer to get onboarded and look at a documentation page about how to write documentation and then write it in this project.

And as I remember, @LysandreJik had sent me docs (which I can't find now) and there it was explained which sections exist and how they can be used.

I'm happy either way, as long as we have somewhere to refer people to, and have them follow those guidelines, rather than them having to look at the rest of the codebase to figure out how to write docs.